options.go

```diff
@@ -31,6 +31,8 @@ type Options struct {
 	// Other options for implementations of the interface
 	// can be stored in a context
 	Context context.Context
+
+	DisableSignalHandler bool
```

Comment:
Too verbose. See the other options. Call it `Signal bool` and add a comment to indicate what its for.